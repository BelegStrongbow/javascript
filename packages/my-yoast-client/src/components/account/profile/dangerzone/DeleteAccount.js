import PropTypes from "prop-types";
import React from "react";
import { injectIntl, intlShape, defineMessages, FormattedMessage } from "react-intl";
import styled from "styled-components";
import defaults from "../../../../config/defaults.json";

// Custom components.
import { RedButton } from "../../../Button";
import { Form } from "../FormElements";

const messages = defineMessages( {
	dangerZone: {
		id: "profile.label.dangerZone",
		defaultMessage: "Danger zone",
	},
	labelDelete: {
		id: "profile.label.delete",
		defaultMessage: "Delete account",
	},
	deletingAccount: {
		id: "profile.deleting",
		defaultMessage: "Deleting your account...",
	},
	deleteAccount: {
		id: "profile.delete",
		defaultMessage: "Delete your account",
	},
	warningMessage: {
		id: "profile.delete.message",
		defaultMessage: "Warning! If you delete your account you lose access to" +
		" your downloads and you will no longer receive updates for any Premium" +
		" plugins you've bought from us.",
	},
} );


const Paragraph = styled.p`
	margin-top: 0.0em;
	margin-bottom: 0.5em;
	font-size: 1em;
	font-weight: 700;
`;

const DeleteButton = styled( RedButton )`
	@media screen and ( max-width: ${ defaults.css.breakpoint.mobile }px ) {
		width: 100%;
	}
`;

class DeleteAccount extends React.Component {
	constructor( props ) {
		super( props );
		this.handleDelete = this.handleDelete.bind( this );
	}

	handleDelete( event ) {
		event.preventDefault();
		this.props.onDeleteProfile();
	}

	/**
	 * Creates the text to be displayed on the save button.
	 *
	 * @returns {string} Text to be used on the submit button.
	 */
	deleteButtonText() {
		return this.props.isDeleting
			? <FormattedMessage id={ messages.deletingAccount.id } defaultMessage={ messages.deletingAccount.defaultMessage } />
			: <FormattedMessage id={ messages.deleteAccount.id } defaultMessage={ messages.deleteAccount.defaultMessage } />;
	}

	/**
	 * Renders the component.
	 *
	 * @returns {ReactElement} The rendered component.
	 */
	render() {
		return <Form onSubmit={ this.handleDelete }>
			<Paragraph>
				<FormattedMessage id={ messages.labelDelete.id } defaultMessage={ messages.labelDelete.defaultMessage } />
			</Paragraph>
			<p>
				<FormattedMessage id={ messages.warningMessage.id } defaultMessage={ messages.warningMessage.defaultMessage } />
			</p>
			<DeleteButton type="submit" disabled={ this.isDeleting }>{ this.deleteButtonText() }</DeleteButton>
		</Form>
		;
	}
}

DeleteAccount.propTypes = {
	intl: intlShape.isRequired,
	onDeleteProfile: PropTypes.func.isRequired,
	isDeleting: PropTypes.bool,
};

DeleteAccount.defaultProps = {
	someProp: "[Some Text]",
	isDeleting: false,
};

export default injectIntl( DeleteAccount );
