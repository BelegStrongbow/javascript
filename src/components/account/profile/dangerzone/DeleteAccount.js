import PropTypes from "prop-types";
import React from "react";
import { injectIntl, intlShape, defineMessages, FormattedMessage } from "react-intl";
import styled from "styled-components";

// Custom components.
import { Page } from "../../../PaperStyles";
import { RedButton } from "../../../Button";


const messages = defineMessages( {
	dangerZone: {
		id: "profile.label.dangerZone",
		defaultMessage: "Danger zone",
	},
	labelDelete: {
		id: "profile.label.delete",
		defaultMessage: "Delete account",
	},
	deletingAccount: {
		id: "profile.deleting",
		defaultMessage: "Deleting your account...",
	},
	deleteAccount: {
		id: "profile.delete",
		defaultMessage: "Delete your account",
	},
	warningMessage: {
		id: "profile.delete.message",
		defaultMessage: "Warning! If you delete your account you lose access to" +
		" your downloads and you will no longer receive updates for any Premium" +
		" plugins you've bought from us.",
	},
} );


const Paragraph = styled.p`
	margin-top: 0.0em;
	margin-bottom: 0.5em;
	font-size: 1em;
	font-weight: 700;
`;

const DeleteButton = styled( RedButton )`
	margin: 1em 0;
`;

class DeleteAccount extends React.Component {

	constructor( props ) {
		super( props );
		this.handleDelete = this.handleDelete.bind( this );
	}

	handleDelete( event ) {
		event.preventDefault();
		this.props.onDeleteProfile();
	}

	/**
	 * Creates the text to be displayed on the save button.
	 *
	 * @returns {string} Text to be used on the submit button.
	 */
	deleteButtonText() {
		return this.props.isDeleting
			? <FormattedMessage id={ messages.deletingAccount.id } defaultMessage={ messages.deletingAccount.defaultMessage}/>
			: <FormattedMessage id={ messages.deleteAccount.id } defaultMessage={ messages.deleteAccount.defaultMessage }/>;
	}

	render() {
		return <Page>
					<form onSubmit={ this.handleDelete }>
						<Paragraph>
							<FormattedMessage id={ messages.labelDelete.id } defaultMessage={ messages.labelDelete.defaultMessage }/>
						</Paragraph>
						<p>
							<FormattedMessage id={ messages.warningMessage.id } defaultMessage={ messages.warningMessage.defaultMessage }/>
						</p>
						<DeleteButton type="submit" disabled={ this.isDeleting }>{ this.deleteButtonText() }</DeleteButton>
					</form>
				</Page>;
	}
}

DeleteAccount.propTypes = {
	intl: intlShape.isRequired,
	onDeleteProfile: PropTypes.func.isRequired,
	isDeleting: PropTypes.bool,
};

DeleteAccount.defaultProps = {
	someProp: "[Some Text]",
};

export default injectIntl( DeleteAccount );
