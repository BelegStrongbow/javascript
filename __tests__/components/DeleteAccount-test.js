import React from 'react';
import { createComponentWithIntl } from "../../utils";

import DeleteAccount from "../../src/components/account/profile/dangerzone/DeleteAccount";

test('The delete account component matches the snapshot', () => {
	const component = createComponentWithIntl(
		<DeleteAccount/>
	);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('The delete profile button is labeled "Deleting your account" when deleting is in process', () => {
	const component = createComponentWithIntl(
		<DeleteAccount isDeleting={ true }/>
	);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Clicking on the "Delete profile" button calls the onDeleteProfile callback.', () => {

	let mockedDeleteCallback = jest.fn();

	// Mock event, so that 'preventDefault' can be called.
	const mockEvent = {
		preventDefault: () => null
	};

	const component = createComponentWithIntl(
		<DeleteAccount onDeleteProfile={ mockedDeleteCallback }/>
	);

	let tree = component.toJSON();

	// Trigger the callback.
	let onSubmit = tree.children[0].props.onSubmit;
	onSubmit(mockEvent);

	expect(mockedDeleteCallback).toHaveBeenCalled();
});