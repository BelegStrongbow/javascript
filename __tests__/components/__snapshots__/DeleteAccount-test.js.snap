// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The delete account component matches the snapshot 1`] = `
.c4 {
  height: 48px;
  margin: 0;
  padding: 0 16px;
  border: 0;
  background-color: #64a60a;
  color: #fff;
  box-shadow: inset 0 -4px 0 rgba(0,0,0,0.2);
  -webkit-filter: drop-shadow(0 2px 4px rgb(0,0,0,0.2));
  filter: drop-shadow(0 2px 4px rgb(0,0,0,0.2));
  border-radius: 4px;
  font: 700 14px/24px "Open Sans",sans-serif;
  cursor: pointer;
  vertical-align: top;
  -webkit-transition: background-color 150ms ease-out;
  transition: background-color 150ms ease-out;
}

.c4:hover,
.c4:focus {
  box-shadow: 0 4px 10px 0 rgba(0,0,0,0.2),inset 0 0 0 100px rgba( 0,0,0,0.1 );
}

.c4:active {
  -webkit-transform: translateY( 1px );
  -ms-transform: translateY( 1px );
  transform: translateY( 1px );
  box-shadow: none;
}

.c3 {
  background-color: #dc3232;
  min-width: 152px;
}

.c0 {
  margin-top: 1.5em;
  margin-bottom: 1em;
}

.c1 {
  margin-top: 0.0em;
  margin-bottom: 0.5em;
  font-size: 1em;
  font-weight: 700;
}

@media screen and ( max-width:800px ) {
  .c2 {
    width: 100%;
  }
}

<form
  className="c0"
  onSubmit={[Function]}
>
  <p
    className="c1"
  >
    <span>
      Delete account
    </span>
  </p>
  <p>
    <span>
      Warning! If you delete your account you lose access to your downloads and you will no longer receive updates for any Premium plugins you've bought from us.
    </span>
  </p>
  <button
    className="c2 c3 c4"
    disabled={undefined}
    type="submit"
  >
    <span>
      Delete your account
    </span>
  </button>
</form>
`;

exports[`The delete profile button is labeled "Deleting your account" when deleting is in process 1`] = `
.c4 {
  height: 48px;
  margin: 0;
  padding: 0 16px;
  border: 0;
  background-color: #64a60a;
  color: #fff;
  box-shadow: inset 0 -4px 0 rgba(0,0,0,0.2);
  -webkit-filter: drop-shadow(0 2px 4px rgb(0,0,0,0.2));
  filter: drop-shadow(0 2px 4px rgb(0,0,0,0.2));
  border-radius: 4px;
  font: 700 14px/24px "Open Sans",sans-serif;
  cursor: pointer;
  vertical-align: top;
  -webkit-transition: background-color 150ms ease-out;
  transition: background-color 150ms ease-out;
}

.c4:hover,
.c4:focus {
  box-shadow: 0 4px 10px 0 rgba(0,0,0,0.2),inset 0 0 0 100px rgba( 0,0,0,0.1 );
}

.c4:active {
  -webkit-transform: translateY( 1px );
  -ms-transform: translateY( 1px );
  transform: translateY( 1px );
  box-shadow: none;
}

.c3 {
  background-color: #dc3232;
  min-width: 152px;
}

.c0 {
  margin-top: 1.5em;
  margin-bottom: 1em;
}

.c1 {
  margin-top: 0.0em;
  margin-bottom: 0.5em;
  font-size: 1em;
  font-weight: 700;
}

@media screen and ( max-width:800px ) {
  .c2 {
    width: 100%;
  }
}

<form
  className="c0"
  onSubmit={[Function]}
>
  <p
    className="c1"
  >
    <span>
      Delete account
    </span>
  </p>
  <p>
    <span>
      Warning! If you delete your account you lose access to your downloads and you will no longer receive updates for any Premium plugins you've bought from us.
    </span>
  </p>
  <button
    className="c2 c3 c4"
    disabled={undefined}
    type="submit"
  >
    <span>
      Deleting your account...
    </span>
  </button>
</form>
`;
